### ISSUE TITLE
Simplify public dashboard logic

### ISSUE THREAD

**Issue by ralienpp** [thingsboard/thingsboard#523]
The current way to mark a dashboard as _public_ is quite tricky. I found no description of how to do this in the manual, so after tinkering with the UI I found the following method to work:

1. Assign my dashboard to `Public`
2. Get the public URL for it

After opening that URL in another browser I saw the dashboard but no data. The browser console log indicated an access error, so I figured that I have to make the device public too. That addressed the issue,  however:

- it is not intuitive that the device should be made public too, there was no error message saying that that is why no data are shown on the dashboard.
- once this is done, anyone can see other public dashboards in the dropdown list, which is not necessarily a good thing.

Ideally, one should be able to mark the dashboard as _public_ with a single action, and without placing all public dashboards in the same namespace.

**Comment by stefanfoulis** [comment]
I think a simple way to make a dashboard public is an important use case. I don't want to make a device public. I want a certain limited view on the data to become public. The dashboard may contain data from multiple devices. Or only some metrics of a device.

I understand though that this is a difficult challenge from a technical perspective.

It would also not necessarily mean the metrics used on a public dashboard are available to all anonymous users. A public dashboard could get a random secret which is becomes part of the shared url. That secret could be used to get an access token that is not bound to a user, instead, it only has scopes for the exact metrics that the dashboard requires.
Basically this would allow semi-public dashboards: no login needed if you have the full url.

**Comment by vparomskiy** [comment]
I know that it will not resolve all your requests but for controlling access to the Device/Asset data you can use EntityViews (https://thingsboard.io/docs/user-guide/entity-views/)
You can configure EntityView to show a limited set of device attributes\telemtetry and a time range (for example last 3 months). 
Once Entity Views are created, you can make them public and use them inside public dashboards. In this case, device still will be private but public users will have access to data from a device.

Second: did you read about the new RBAC model in ThingsBoard 2.3? Maybe it will help to implement your case: https://thingsboard.io/docs/user-guide/rbac/

**Comment by stefanfoulis** [comment]
@vparomskiy thanks, I did not know about EntityViews. It looks like many of the building blocks to implement something similar to my suggestion already exist. Cool!
